ui/src/app/pages/admin/admin-institution-edit.tsx

```diff
@@ -0,0 +1,411 @@
+      {label: 'Other', value: OrganizationType.OTHER}
+    ];
+    return options;
+  }
+
+  get institutionAgreementTypeOptions() {
+    const options = [
+      {label: 'Master', value: DuaType.MASTER},
+      {label: 'Individual', value: DuaType.RESTRICTED}
+    ];
+    return options;
+  }
+
+  // Filter out empty line or empty email addresses like <email1>,,<email2>
+  // Confirm each email is a valid email using validate.js
+  validateEmailAddresses() {
+    const invalidEmailAddress = [];
+    const {emailAddresses} = this.state.institution;
+    this.state.institution.emailAddresses = emailAddresses.filter(
+      emailAddress => {
+        return emailAddress !== '' || !!emailAddress;
+      });
+
+    this.state.institution.emailAddresses.map(emailAddress => {
+      const errors = validate({
+        emailAddress
+      }, {
+        emailAddress: {email: true}
+      });
+      if (errors && errors.emailAddress && errors.emailAddress.length > 0) {
+        invalidEmailAddress.push(emailAddress);
+      }
+    });
+    this.setState({invalidEmailAddress: invalidEmailAddress.length > 0});
+    if (invalidEmailAddress.length > 0) {
+      const errMessage = 'Following Email Addresses are not valid : ' + invalidEmailAddress.join(' , ');
+      this.setState({invalidEmailAddressMsg: errMessage});
+    }
+    return;
+  }
+
+  // Filter out empty line or empty email addresses like <email1>,,<email2>
+  // Confirm each email domain matches with regex
+  validateEmailDomain() {
+    const invalidEmailDomain = [];
+    const {emailDomains} = this.state.institution;
+    this.state.institution.emailDomains =  emailDomains.filter(emailDomain => emailDomain);
+    this.state.institution.emailDomains.map(emailDomain => {
+      const errors = validate({
+        emailDomain
+      }, {
+        emailDomain: {format: {pattern: /[a-zA-z\-\.]+[.][a-zA-Z]+/i}}
+      });
+      if (errors && errors.emailDomain && errors.emailDomain.length > 0) {
+        invalidEmailDomain.push(emailDomain);
+      }
+    });
+    this.setState({invalidEmailDomain: invalidEmailDomain.length > 0});
+    if (invalidEmailDomain.length > 0) {
+      const errMessage = 'Following Email Domains are not valid : ' + invalidEmailDomain.join(' , ');
+      this.setState({invalidEmailDomainsMsg: errMessage});
+    }
+    return;
+  }
+
+  setEmailDomain(emailDomains, attribute) {
+    const emailDomainList = emailDomains.split(/[,\n]+/);
+    this.setState(fp.set(['institution', attribute], emailDomainList));
+  }
+
+  // Check if the fields have not been edited
+  fieldsNotEdited() {
+    return (this.state.isAddInstitution && !this.fieldsNotEditedAddInstitution)
+        || (institutionToEdit && this.fieldsNotEditedEditInstitution);
+  }
+
+  get fieldsNotEditedAddInstitution() {
+    const {institution} = this.state;
+    return institution.displayName || institution.organizationTypeOtherText ||
```

Comment:
missing userInstructions

Reply:
User Instruction is optional

Reply:
that is also true for organizationTypeOtherText when type is not Other

Reply:
After taking another look i am adding userInstruction check as its for fieldNotEdit back button not required field check. I feel i can remove organizationTypeOtherText as it is redundant